lib/puppet/node.rb

```diff
@@ -27,7 +27,10 @@ def initialize_from_hash(data)
     @name       = data['name']       || (raise ArgumentError, _("No name provided in serialized data"))
     @classes    = data['classes']    || []
     @parameters = data['parameters'] || {}
-    @environment_name = data['environment']
+    env_name = data['environment']
+    env_name = env_name.intern unless env_name.nil?
+    @environment_name = env_name
+    environment = env_name
```

Comment:
What if you run `puppet agent` and the agent is configured with an environment name `foo` but the environment directory doesn't exist? Or if you run `puppet resource` and puppet.conf references a non-existent directory?